### PR TITLE
Updating releases notes and image

### PR DESCRIPTION
**Pull request by erikmarr**
## Purpose of this pull request

This pull request (PR) updates the release notes for product recommendations as well as an outdated image.

## Affected DevDocs pages

-  https://devdocs.magento.com/recommendations/release-notes.html
-  https://devdocs.magento.com/recommendations/verify.html
-  https://devdocs.magento.com/release/availability.html

### CODE REVIEW

**Review by misha-kotov — CHANGES_REQUESTED** [review]
Release notes look good. 
Compatibility: 
3.0.8 should be `compatible` across the board - looks like there are still some `supported` currently. same for 3.0.6

**Review by misha-kotov — APPROVED** [review]

File: src/recommendations/release-notes.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ The release notes include:
 -  {:.new}New features
 -  {:.fix}Fixes and improvements
 
+## 3.10.0 of magento/product-recommendations for Magento Commerce 2.4.0
+
+-  {:.new}Added ability to [resync your catalog to SaaS services via command line]({{ page.baseurl }}/config-guide/cli/config-cli-subcommands-catalog-sync.html)
```
Comment (keharper) [inline src/recommendations/release-notes.md:20]:
```suggestion
-  {:.new}Added the ability to [resync your catalog to SaaS services via command line]({{ page.baseurl }}/config-guide/cli/config-cli-subcommands-catalog-sync.html)
```

**Review by keharper — APPROVED** [review]
One tiny change

**Comment by erikmarr** [comment]
running tests

**Comment by erikmarr** [comment]
running tests

**Pull request merged**